src/internal/connector/graph_connector_helper_test.go

```diff
@@ -643,20 +646,42 @@ func compareOneDriveItem(
 	expected map[string][]byte,
 	item data.Stream,
 ) {
+	name := item.UUID()
+
 	expectedData := expected[item.UUID()]
 	if !assert.NotNil(t, expectedData, "unexpected file with name %s", item.UUID) {
 		return
 	}
 
-	// OneDrive items are just byte buffers of the data. Nothing special to
-	// interpret. May need to do chunked comparisons in the future if we test
-	// large item equality.
 	buf, err := io.ReadAll(item.ToReader())
 	if !assert.NoError(t, err) {
 		return
 	}
 
-	assert.Equal(t, expectedData, buf)
+	if strings.HasSuffix(name, onedrive.MetaFileSuffix) || strings.HasSuffix(name, onedrive.DirMetaFileSuffix) {
+		var (
+			itemMeta     onedrive.Metadata
+			expectedMeta onedrive.Metadata
+		)
+
+		err := json.Unmarshal(buf, &itemMeta)
+		assert.Nil(t, err)
+
+		err = json.Unmarshal(expectedData, &expectedMeta)
+		assert.Nil(t, err)
+		assert.Equal(t, len(expectedMeta.Permissions), len(itemMeta.Permissions), "number of permissions after restore")
+
+		for i, p := range expectedMeta.Permissions {
+			assert.Equal(t, p.Email, itemMeta.Permissions[i].Email)
+			assert.Equal(t, p.Roles, itemMeta.Permissions[i].Roles)
+			assert.Equal(t, p.Expiration, itemMeta.Permissions[i].Expiration)
+		}
```

Comment:
this may need to be more robust in the future. It seems like it currently assumes a fixed order for permissions that is common between both the expected and actual permissions

Reply:
Let me add a note in there for now. As of now the tests are only using single permissions as we have only one additional user available `CORSO_SECONDARY_M365_TEST_USER_ID`.